pkg/virt-controller/watch/vmi.go

```diff
@@ -490,32 +490,24 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 }
 
 // isPodReady treats the pod as ready to be handed over to virt-handler, as soon as all pods except
-// the compute pod are ready. That includes kubevirt-infra and sidecars.
+// the compute pod are ready.
 func isPodReady(pod *k8sv1.Pod) bool {
 	if isPodDownOrGoingDown(pod) {
 		return false
 	}
-	infraContainerExists := false
-	for _, containerStatus := range pod.Status.ContainerStatuses {
-		if containerStatus.Name == "kubevirt-infra" {
-			infraContainerExists = true
-			break
-		}
-	}
+
 	for _, containerStatus := range pod.Status.ContainerStatuses {
 		// If there is a kubevirt-infra pod, we should not check "compute" here.
 		// Otherwise all containers including "compute" should be "ready before we go on.
```

Comment:
This probably needs a different explanation, specifically why do we check `State.Running == nil` for the compute container.

Reply:
Yes, updated. The readiness probe on the compute container is used to indicate VMI readiness therefore we can't wait for it to become ready since we have to start the VMI for it. Instead it just checks if the container was started.